File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> hash;
        int left=0;
        int right=0;
        int max_len=0;
        while(right<s.length()){
            if(hash.find(s[right])!=hash.end()){
                hash.erase(s[left]);
                left++;
            }
            else{
                hash.insert(s[right]);
                right++;
                int len=right-left;
                max_len=max(max_len,len);
            }
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sub=nums[0]
        curr_sum=nums[0]

        left,right=1,1
        while(right<len(nums)):
            if curr_sum<0:
                curr_sum=nums[right]
                left=right
            else:
                curr_sum+=nums[right]
            right+=1
            max_sub=max(max_sub,curr_sum)

        return max_sub